cmake/Platform.cmake

```diff
@@ -22,4 +22,34 @@ if (MSVC)
     # Suppress warnings: "Prefer enum class over enum" (Enum.3)
 
     add_compile_options(/wd4267 /wd26812)
+
+    if(NOT QL_MSVC_DISABLE_PARALLEL_BUILDS)
+        # enable parallel builds
+        add_compile_options(/MP)
+    endif()
+
+    # warning level 3 and all warnings as errors
+    add_compile_options(/W3 /WX)
+
+    ### Avoid level 3 warnings for the time being.
+    ### However they should be fixed in the long run.
+
+    # caused by ql\time\date.cpp: warning C4996: 'localtime': This function or variable may be unsafe. Consider using localtime_s instead. 
+    add_compile_definitions(_CRT_SECURE_NO_DEPRECATE)
+
+    # caused by macro redefinition 'M_PI' etc. e.g. in ql\methods\finitedifferences\solvers\fdmbackwardsolver.cpp
+    add_compile_options(/wd4005)
+
+    # prevent warnings when using /std:c++17
+    if(CMAKE_CXX_STANDARD GREATER 14)
+        # caused by the QL_ENABLE_THREAD_SAFE_OBSERVER_PATTERN
+        add_compile_definitions(BOOST_UNORDERED_USE_ALLOCATOR_TRAITS=2)
+        
+        # caused by #include <boost/numeric/ublas/matrix.hpp> 
+        # in e.g. ql\experimental\finitedifferences\fdmdupire1dop.cpp
+        add_compile_definitions(_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING)
+
+        # caused by ql/math/functional.hpp (line 271) compiling test-suite\distributions.cpp
+        add_compile_definitions(_SILENCE_CXX17_ADAPTOR_TYPEDEFS_DEPRECATION_WARNING)
+    endif()
```

Comment:
Suppose I want a C++17 build in debug mode specifically to see these warnings?  All of this should be removed.   Focus on fixing the source code and not twisting the building system around observed warnings.

Reply:
> Suppose I want a C++17 build in debug mode specifically to see these warnings?

Be pragmatic and comment out the part you are working on. Remove it once you have fixed the issue for the warning.